llvm/lib/Transforms/ObjCARC/ObjCARCContract.cpp

```diff
@@ -730,6 +740,7 @@ INITIALIZE_PASS_END(ObjCARCContractLegacyPass, "objc-arc-contract",
 void ObjCARCContractLegacyPass::getAnalysisUsage(AnalysisUsage &AU) const {
   AU.addRequired<AAResultsWrapperPass>();
   AU.addRequired<DominatorTreeWrapperPass>();
+  AU.addPreserved<DominatorTreeWrapperPass>();
```

Comment:
Can you also add AAResultsWrapper and BasicAAWrapperPass here? I think that should avoid some more pipeline diffs.